modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/rest/resource/releases/SW360Release.java

```diff
@@ -192,26 +201,69 @@ public void setCopyrights(String copyrights) {
     }
 
     public boolean shareIdentifier(SW360Release releaseCompare) {
-        return this.componentId.equals(releaseCompare.getComponentId())
-                && this.name.equals(releaseCompare.getName())
-                && this.version.equals(releaseCompare.getVersion());
+        return this.name.equals(Optional.of(releaseCompare.getName()).orElse(""))
+                && this.version.equals(Optional.of(releaseCompare.getVersion()).orElse(""));
+    }
+
+    public SW360Release mapExternalIdsOnVariable() {
+        externalIds.forEach((key, value) -> {
+            switch (key) {
+                case SW360Attributes.RELEASE_EXTERNAL_ID_FLICENSES:
+                    finalLicense = value;
+                    break;
+                case SW360Attributes.RELEASE_EXTERNAL_ID_DLICENSES:
+                    declaredLicense = value;
+                    break;
+                case SW360Attributes.RELEASE_EXTERNAL_ID_OLICENSES:
+                    observedLicense = value;
+                    break;
+                case SW360Attributes.RELEASE_EXTERNAL_ID_OREPO:
+                    releaseTagUrl = value;
+                    break;
+                case SW360Attributes.RELEASE_EXTERNAL_ID_SWHID:
+                    softwareHeritageId = value;
+                    break;
+                case SW360Attributes.RELEASE_EXTERNAL_ID_CHANGESTATUS:
+                    changeStatus = value;
+                    break;
+                case SW360Attributes.RELEASE_EXTERNAL_ID_COPYRIGHTS:
+                    copyrights = value;
+                    break;
+                default:
+                    if (key.startsWith(SW360Attributes.RELEASE_EXTERNAL_ID_HASHES)) {
+                        if(hashes == null) {
+                            hashes = Collections.singleton(value);
+                        } else {
+                            hashes.add(value);
+                        }
+                    } else if (SW360CoordinateKeysToArtifactCoordinates.getValues().contains(key)) {
+                        if(coordinates == null) {
+                            coordinates = new HashMap<String, String>() {{
```

Comment:
I think this way of initializing is deprecated (or at least it is ugly).
I would even prefer:
```suggestion
                            coordinates = new HashMap<String, String>();
                            coordinates.put(key, value);
```

(and even more I would prefer an immutable map, but that is a completely different story ;) )